network/server.go

```diff
@@ -329,10 +330,17 @@ func (s *Server) checkPeerConnections() {
 
 		if s.numPeers() < MinimumPeerConnections {
 			if s.config.NoDiscover || !s.bootnodes.hasBootnodes() {
-				// TODO: dial peers from the peerstore
+				// dial unconnected peer
+				randPeer := s.GetRandomPeer()
+				if randPeer != nil && !s.IsConnected(*randPeer) {
+					s.addToDialQueue(s.GetPeerInfo(*randPeer), common.PriorityRandomDial)
+				}
 			} else {
+				// dial random unconnected bootnode
 				randomNode := s.GetRandomBootnode()
-				s.addToDialQueue(randomNode, common.PriorityRandomDial)
+				if randomNode != nil {
+					s.addToDialQueue(randomNode, common.PriorityRandomDial)
```

Comment:
You can inline this

Reply:
```go
				if randomNode := s.GetRandomBootnode(); randomNode != nil {
					s.addToDialQueue(randomNode, common.PriorityRandomDial)
				}
```

Reply:
Good catch 🎖️  Fixed in: 1531d1aa1f19ed65d2448230e0b026c1488cac2b